packages/events-service/src/controllers/event/list.ts

```diff
@@ -0,0 +1,54 @@
+import { Response } from 'express'
+import { stringifyError } from '@reapit/node-utils'
+import { between, equals } from '@aws/dynamodb-expressions'
+import { logger } from '../../core/logger'
+import { db } from '../../core/db'
+import { AppRequest, ReqUser } from '../../types/request'
+import { DecoratedEvents } from '../../services/decorated-events'
+import { EventListQuery } from '../../types/event-list-query'
+import { EventStatus } from '../../schemas/event-status.schema'
+
+const getOutstandingEvents = async (query: EventListQuery, user: ReqUser): Promise<EventStatus[]> => {
+  if (user?.clientCode !== query.clientCode) {
+    throw new Error('Unauthorized clientCode')
+  }
+
+  const keyCondition = {
+    clientCode: query.clientCode,
+    eventCreatedAt: between(query.dateFrom, query.dateTo),
+  }
+
+  const queryConditons = {
+    indexName: 'EventStatusesByClientCodeAndEventCreatedDate',
+    filter: {
+      subject: 'status',
+      ...equals('outstanding'),
+    },
+  }
+
+  const iterator = db.query(EventStatus, keyCondition, queryConditons)
+  const records = []
+  for await (const record of iterator) records.push(record)
```

Comment:
Looks fancy 😁 Not sure I understand what's going on - could maybe use a comment...